pkgs/development/python-modules/cfn-flip/default.nix

```diff
@@ -1,25 +1,39 @@
-{ lib, buildPythonPackage, fetchPypi, six, pyyaml, click, pytestrunner }:
+{ lib
+, buildPythonPackage
+, fetchPypi
+
+# pythonPackages
+, click
+, pyyaml
+, six
+}:
 
 buildPythonPackage rec {
   pname = "cfn-flip";
-  version = "1.1.0.post1";
+  version = "1.2.2";
 
   src = fetchPypi {
     pname = "cfn_flip";
     inherit version;
-    sha256 = "16r01ijjwnq06ax5xrv6mq9l00f6sgzw776kr43zjai09xsbwwck";
+    sha256 = "05n5vvc5lqdzssgyb8kvb23byavs84ww4vbylaj13kakxxn05pcd";
   };
 
-  propagatedBuildInputs = [ six pyyaml click ];
-  nativeBuildInputs = [ pytestrunner ];
+  propagatedBuildInputs = [
+    click
+    pyyaml
+    six
+  ];
 
-  # No tests in Pypi
+  # No tests in PyPi
   doCheck = false;
```

Comment:
it would be preferable to checkout from github to get some unittests. Want to ensure that the package is still usable under normal circumstances

Reply:
@jonringer 

Applied the following changes, test are exactly the same as in the tox.ini file for the source repository

```diff
diff --git a/pkgs/development/python-modules/cfn-flip/default.nix b/pkgs/development/python-modules/cfn-flip/default.nix
index 12ebbfe076a..4b9ce6e9027 100644
--- a/pkgs/development/python-modules/cfn-flip/default.nix
+++ b/pkgs/development/python-modules/cfn-flip/default.nix
@@ -1,9 +1,12 @@
-{ lib
-, buildPythonPackage
-, fetchPypi
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
 
 # pythonPackages
 , click
+, pytest
+, pytestcov
+, pytestrunner
 , pyyaml
 , six
 }:
@@ -12,10 +15,11 @@ buildPythonPackage rec {
   pname = "cfn-flip";
   version = "1.2.2";
 
-  src = fetchPypi {
-    pname = "cfn_flip";
-    inherit version;
-    sha256 = "05n5vvc5lqdzssgyb8kvb23byavs84ww4vbylaj13kakxxn05pcd";
+  src = fetchFromGitHub {
+    owner = "awslabs";
+    repo = "aws-cfn-template-flip";
+    rev = version;
+    sha256 = "05fk725a1i3zl3idik2hxl3w6k1ln0j33j3jdq1gvy1sfyc79ifm";
   };
 
   propagatedBuildInputs = [
@@ -24,8 +28,20 @@ buildPythonPackage rec {
     six
   ];
 
-  # No tests in PyPi
-  doCheck = false;
+  checkInputs = [
+    pytest
+    pytestcov
+    pytestrunner
+  ];
+
+  checkPhase = ''
+    py.test \
+      --cov=cfn_clean \
+      --cov=cfn_flip \
+      --cov=cfn_tools \
+      --cov-report term-missing \
+      --cov-report html
+  '';
 
   meta = with lib; {
     description = "Tool for converting AWS CloudFormation templates between JSON and YAML formats";
```